pkg/reconciler/ingress/ingress_test.go

```diff
@@ -247,61 +252,85 @@ func withAnnotations(ans map[string]string) ingressCreationOption {
+			Splits: []netv1alpha1.IngressBackendSplit{
+				{
+					IngressBackend: v1alpha1.IngressBackend{
+						ServiceName:      serviceName,
+						ServiceNamespace: namespace,
+						ServicePort:      intstr.FromInt(80),
+					},
+					Percent:       int(100),
+					AppendHeaders: map[string]string{"K-Original-Host": testHost},
+				}},
+		},
+			{
+				RewriteHost: getClusterLocalDomain(producerServiceName, knativeTesting),
+				Splits: []netv1alpha1.IngressBackendSplit{
+					{
+						Percent: 100,
+						IngressBackend: netv1alpha1.IngressBackend{
+							ServiceNamespace: namespace,
+							ServiceName:      name + asyncSuffix,
+							ServicePort:      intstr.FromInt(80),
+						},
+					},
+				},
+				AppendHeaders: map[string]string{asyncOriginalHost: getClusterLocalDomain(name, defaultNamespace)},
 			},
-			Percent: int(100),
 		})
-		theRules := []v1alpha1.IngressRule{}
-		for _, rule := range ing.Spec.Rules {
-			newRule := rule
-			newPaths := make([]v1alpha1.HTTPIngressPath, 0)
-			appendHeaders := make(map[string]string)
-			appendHeaders["Async-Original-Host"] = getClusterLocalDomain(ingName, defaultNamespace)
-			if isAlwaysAsync {
-				for _, path := range rule.HTTP.Paths {
-					defaultPath := path
-					defaultPath.Splits = splits
-					defaultPath.RewriteHost = getClusterLocalDomain(producerServiceName, "knative-testing")
-					defaultPath.AppendHeaders = appendHeaders
-					if path.Headers == nil {
-						path.Headers = map[string]v1alpha1.HeaderMatch{preferHeaderField: {Exact: preferSyncValue}}
-					} else {
-						path.Headers[preferHeaderField] = v1alpha1.HeaderMatch{Exact: preferSyncValue}
-					}
-					newPaths = append(newPaths, path, defaultPath)
-					newRule.HTTP.Paths = newPaths
-					theRules = append(theRules, newRule)
-				}
-			} else {
-				newPaths = append(newPaths, v1alpha1.HTTPIngressPath{
-					Headers:       map[string]v1alpha1.HeaderMatch{preferHeaderField: {Exact: preferAsyncValue}},
-					Splits:        splits,
-					RewriteHost:   getClusterLocalDomain(producerServiceName, "knative-testing"),
-					AppendHeaders: appendHeaders,
-				})
-				newPaths = append(newPaths, newRule.HTTP.Paths...)
-				newRule.HTTP.Paths = newPaths
-				theRules = append(theRules, newRule)
-			}
-		}
-		ing.Spec.Rules = theRules
+	} else {
+		thePaths = append([]netv1alpha1.HTTPIngressPath{{
+			RewriteHost: getClusterLocalDomain(producerServiceName, knativeTesting),
+			Headers:     map[string]v1alpha1.HeaderMatch{preferHeaderField: {Exact: preferAsyncValue}},
+			Splits: []netv1alpha1.IngressBackendSplit{
+				{
+					IngressBackend: v1alpha1.IngressBackend{
+						ServiceName:      name + asyncSuffix,
+						ServiceNamespace: namespace,
+						ServicePort:      intstr.FromInt(80),
+					},
+					Percent: int(100),
+				}},
+			AppendHeaders: map[string]string{
+				asyncOriginalHost: getClusterLocalDomain(name, defaultNamespace),
+			},
+		},
+			{Splits: []netv1alpha1.IngressBackendSplit{
```

Comment:
formatting odd here aswell, did GoFmt not run?

Reply:
for some reason this looks a bit odd on github. I'm using vscode, and it looks fine -- also ran gofmt & goimports on the file.